silq/parameters/acquisition_parameters.py

```diff
@@ -1858,17 +1860,19 @@ def ESR_frequencies(self, ESR_frequencies: List):
         self.ESR['ESR_pulses'] = updated_ESR_pulses
 
     def analyse(self, traces: Dict[str, Dict[str, np.ndarray]] = None):
-        """Analyse flipping events between nuclear states
+        """Analyse flipping events between nuclear states and determine nuclear state
 
         Returns:
             (Dict[str, Any]): Dict containing:
 
-            :results_read (dict): `analyse_traces` results for each read
+            * **results_read** (dict): `analyse_traces` results for each read
               trace
-            :up_proportions_{idx} (np.ndarray): Up proportions, the
+            * **up_proportions_{idx}** (np.ndarray): Up proportions, the
               dimensionality being equal to ``NMRParameter.samples``.
               ``{idx}`` is replaced with the zero-based ESR frequency index.
-            :Results from `analyse_flips`. These are:
+            * **state_probability_0_{idx}**, **state_probability_1_{idx}** (np.ndarray):
+              respectively up, down nuclear spin state proportions when reading-out ESR1 transition
```

Comment:
This isn't generic, ESR1 shouldn't be mentioned here because the way the NMR parameter works is to measure an up-proportion for arbitrary ESR pulses.

For example, one might measure with ESR_frequencies:
`ESR_frequencies = [[ESR2, ESR1], ESR1]`
or
`ESR_frequencies = [ESR1, ESR2, ESR1]`
or
`ESR_frequencies = [ESR1, ESR2, ESR3]`

The first example shows a result from 2 ESR stimuli, one which addressed both ESR1 + ESR2 and the other which addresses ESR1. The first does not give you a simple 'state_probability'!
The last example shows that you can measure more than 2 ESR frequencies, e.g. for antimony this is inevitable.

So I have 2 suggestions:
- Make `state_probability_0_{idx}` more generic, and matched with the number of ESR_frequencies
I would suggest renaming this to something like `success_probability` which inherits its name from standard binomial distribution nomenclature. You can also include 'failure_probability`, but this is redundant (if you do, make sure you have one threshold be inclusive! i.e. use >= or <= ).
In this way `success_probability` would simply replace `state_probability_1`.
- Include ESR_frequencies in the dataset as a setpoint, if it isn't already.

Reply:
"Include ESR_frequencies in the dataset as a setpoint, if it isn't already." - do you mean by this that I should add saving of different ESR frequencies that we are using for readout?

Reply:
Sorry, I realise that was beyond the scope of this PR, so I take back that suggestion, I will deal with that in a separate PR in the future.

Reply:
ok, cool